pkg/topology/kademlia/kademlia.go

```diff
@@ -570,6 +574,25 @@ func (k *Kad) manage() {
 	}
 }
 
+func (k *Kad) pruneOversaturatedBins(depth uint8) {
+
+	for i := uint8(0); i < depth; i++ {
+
+		peers := k.connectedPeers.BinPeers(i)
+		lenPeers := len(peers)
+
+		if lenPeers > saturationPeers {
```

Comment:
use overSaturationPeers instead of saturationPeers so its only pruned if oversaturation limit is breached